sdk/program/src/bignum.rs

```diff
@@ -0,0 +1,784 @@
+//! @brief Solana Rust-based BigNumber for bpf-programs
+
+use borsh::{BorshDeserialize, BorshSchema, BorshSerialize};
+use std::fmt;
+
+#[repr(C)]
+#[derive(
+    BorshSerialize,
+    BorshDeserialize,
+    BorshSchema,
+    Clone,
+    Eq,
+    PartialEq,
+    Ord,
+    PartialOrd,
+    Hash,
+    AbiExample,
+)]
+pub struct BigNumber {
+    value: Vec<u8>,
+    negative: bool,
+}
+
+// pub struct BigNumber(u64);
+
+impl BigNumber {
+    /// Returns a BigNumber with initial value of 0
+    pub fn new() -> Self {
+        Self {
+            negative: false,
+            value: vec![0u8],
+        }
+    }
+
+    /// Returns the size, in bytes, of BigNum. Typically used in
+    /// assessing buffer size needed to fetch the bytes for serialization, etc.
+    pub fn size_in_bytes(&self) -> usize {
+        self.value.len()
+    }
+
+    /// Returns a BigNumber with initial value set to a u32 value
+    pub fn from_u32(val: u32) -> Self {
```

Comment:
Maybe return result?  Better than `unwrap` and mimics the underlying `BigNum`.

Reply:
Most BigNumber calls now return `Result<..>`

Reply:
Fixed in a156563ebf54e8934b207cb12cb2d8aefb83b93d